sim/dex.ts

```diff
@@ -1160,8 +1182,9 @@ class ModdedDex {
 			if (j < 0) return null;
 			const ability = buf.substring(i, j);
 			const template = dexes['base'].getTemplate(set.species);
-			// @ts-ignore
-			set.ability = (template.abilities && ['', '0', '1', 'H'].includes(ability) ? template.abilities[ability || '0'] : ability);
+			set.ability = (template.abilities && ['', '0', '1', 'H'].includes(ability)
+				// @ts-ignore
+				? template.abilities[ability || '0'] : ability);
```

Comment:
```js
			set.ability = ['', '0', '1', 'H', 'S'].includes(ability) ?
				template.abilities[ability as '0' || '0'] || (ability === '' ? '' : '!!!ERROR!!!') :
				ability;
```

is the corresponding client code, incidentally.

The server code can probably be updated to match.